release/release/check_prereqs.clj

```diff
@@ -0,0 +1,83 @@
+(ns release.check-prereqs
+  (:require [clojure.string :as str]
+            [environ.core :as env]
+            [release.common :as c]))
+
+(def ^:private required-commands
+  ["yarn" "aws" "docker" "java"])
+
+(defn- check-for-required-commands []
+  (c/step "Verify required external commands are available"
+    (doseq [cmd required-commands]
+      (c/step (format "Verify command %s is available" (pr-str cmd)))
+      (loop []
+        (let [result (try
+                       (c/sh "which" cmd)
+                       (catch Throwable _
+                         :fail))]
+          (when (= result :fail)
+            (printf "The %s command is not available locally. Please install it and press any key to continue, or Ctrl-C to quit."
+                    (pr-str cmd))
+            (flush)
+            (read-line)
+            (recur)))))
+    (c/announce "All required external commands are available.")))
+
+(def ^:private required-env-vars
+  (filter
+   some?
+   [:github-token
+    :dockerhub-username
+    :dockerhub-password
```

Comment:
I'm not sure if this is intentional or not, but the other scripts try to `docker login` with these credentials. I actually prefer if that's taken care of **outside** of this script, which means these variables are unused / unnecessary.